x-pack/plugins/maps/public/connected_components/add_layer_panel/view.tsx

```diff
@@ -68,13 +68,14 @@ export class AddLayerPanel extends Component<Props, State> {
   };
 
   _onWizardSelect = (layerWizard: LayerWizard) => {
-    const layerSteps = [
-      ...(layerWizard.prerequisiteSteps ? layerWizard.prerequisiteSteps : []),
-      {
-        id: ADD_LAYER_STEP_ID,
-        label: ADD_LAYER_STEP_LABEL,
-      },
-    ];
+    const layerSteps = layerWizard.prerequisiteSteps
+      ? layerWizard.prerequisiteSteps
+      : [
+          {
+            id: ADD_LAYER_STEP_ID,
+            label: ADD_LAYER_STEP_LABEL,
+          },
+        ];
```

Comment:
If all layers ultimately end in an "Add layer step", what's the point in overriding this with prereq steps that include an add layer step vs. always including it here as we did before?

Reply:
I did this so ClientFileCreateSourceEditor would know when the final step without needing to know about ADD_LAYER_STEP_ID since ADD_LAYER_STEP_ID seemed like an internal detail in AddLayerPanel.

I will revert this change and instead, pass isOnFinalStep to renderWizard since that is the information that is needed by ClientFileCreateSourceEditor. That way, ADD_LAYER_STEP_ID is the final step even when prerequisiteSteps are provided.